src/BenchmarkDotNet.Toolchains.Uap/Toolchains/Uap/DevicePortalApiWrapper.cs

```diff
@@ -270,32 +307,40 @@ private static Task AsTask(CancellationToken cancellationToken)
         internal string[] StopListening(ClientWebSocket ws)
         {
             List<string> ret = new List<string>();
+            StringBuilder sb = new StringBuilder();
             var buffer = new byte[10240];
             for (;;)
             {
-                CancellationTokenSource cts = new CancellationTokenSource();
-                var resTask = ws.ReceiveAsync(new ArraySegment<byte>(buffer), cts.Token);
-                cts.CancelAfter(TimeSpan.FromSeconds(10));
-                try
+                using (CancellationTokenSource cts = new CancellationTokenSource())
                 {
-                    resTask.Wait();
-                    var str = Encoding.UTF8.GetString(buffer);
-                    var @object = SimpleJson.SimpleJson.DeserializeObject<RootObject>(str);
-                    ret.AddRange(@object.Events.Select(x => x.StringMessage));
-                }
-                catch (AggregateException ex)
-                {
-                    if (ex.InnerException is OperationCanceledException)
+                    var resTask = ws.ReceiveAsync(new ArraySegment<byte>(buffer), cts.Token);
+                    cts.CancelAfter(TimeSpan.FromMinutes(1));
+                    try
                     {
-                        break;
+                        var res = resTask.Result;
+                        var str = Encoding.UTF8.GetString(buffer, 0, res.Count);
+                        sb.Append(str);
+                        if (res.EndOfMessage)
+                        {
+                            var @object = SimpleJson.SimpleJson.DeserializeObject<RootObject>(sb.ToString());
+                            ret.AddRange(@object.Events.Select(x => x.StringMessage));
+                            sb.Clear();
+                        }
                     }
-                    else
+                    catch (AggregateException ex)
```

Comment:
you can simplify the code by using "when" keyword:

catch (AggregateException ex) when (ex.InnerException is OperationCanceledException)
{
   // empty on purpose
}